### PR TITLE
Genome alignment hisat2@0.2.0

### PR DESCRIPTION
**Pull request by akahles**

### CODE REVIEW

**Review by lindaxiang — COMMENTED** [review]
All looks good to me.

File: genome-alignment-hisat2/alignHISAT2.py [hunk 1]
```diff
@@ -78,92 +186,140 @@ def main():
...
+            sys.exit("Error: %s. HISAT2 failed.\n" % e)
+
+    ### merge read group bams to sample level bam
+    if len(read_group_ids) > 1:
+        cmd = ['samtools', 'merge', args.sample + '_Aligned.out.bam']
+        for rg_id in read_group_ids:
+            cmd.append(args.sample + '_RG_' + rg_id + '_Aligned.out.bam')
+        cmd.extend(['&&', 'rm ' + args.sample + '_RG_*bam'])
+    else:
+        cmd = ['mv', args.sample + '_RG_' + read_group_ids[0] + '_Aligned.out.bam', args.sample + '_Aligned.out.bam']
```
[the remark this answers is not in the document]
Comment (akahles) [reply genome-alignment-hisat2/alignHISAT2.py:308]: Sure, should I make the same change also for the STAR pipeline? Then the naming would be consistent.

[the remark this answers is not in the document]
Comment (akahles) [reply genome-alignment-hisat2/alignHISAT2.py]: Added a `.hisat2` tag to the output file name.

File: genome-alignment-hisat2/alignHISAT2.py [hunk 2]
```diff
@@ -78,92 +186,140 @@ def main():
...
+
+    ### sort output by coordinate
+    bam = args.sample + '_Aligned.out.bam'
+    subprocess.run(f'samtools sort -o {bam}.sorted {bam} && mv {bam}.sorted {bam}', shell=True, check=True)
+
+    ### collect novel splice sites
+    subprocess.run(f'sort -k1,1 -k2,3n -k4,4 -u {args.sample}_RG_*.novel_splicesites.txt > {args.sample}.novel_splicesites.txt && rm {args.sample}_RG_*.novel_splicesites.txt', shell=True, check=True)
+
+    ### bundle logs into tarball
+    subprocess.run(f'tar -czf {args.sample}.all_logs.supplement.tar.gz *metrics.txt *summary.txt align.log', shell=True, check=True)
```
[the remark this answers is not in the document]
Comment (akahles) [reply genome-alignment-hisat2/alignHISAT2.py]: Added a .hisat2 tag to the output file name.

**Comment by akahles** [comment]
I did not add an optional `--tmpdir` for HISAT2, as it either does not need it or does not seem to support it.

**Review by lindaxiang — APPROVED** [review]
LGTM

**Pull request merged**